### DISCUSSION TITLE
Engine open mode not valid error

### DISCUSSION THREAD

**Discussion by hr87** [ornladios/ADIOS2#4152] [Q&A]
Hello,

I am trying to append to an existing file, but I get the following exception
```
terminate called after throwing an instance of 'std::invalid_argument'
  what():  [Fri Apr 26 08:38:41 2024] [ADIOS2 EXCEPTION] <Core> <Engine> <CheckOpenModes> : Engine open mode not valid for  for variable intensities, in call to Variable<T>::Span Put
```

Weird thing, in my unit tests, append works as intended, but in the actual code if throws the error. The file is written by the same code, I first write the beginning, then I change the mode to append, recompile and get this error. (I have a better way to do this for actual runs, but that involves restart, and currently I am trying to get the simplest possible example).

Note that the main program is Fortran, calling a C++ library, that then calls adios2 for IO. The Fortran main program also has two separate Adios2 writers, that write to different files each. There is no overlap between any of the adios implementations.

**Comment by eisenhauer** [comment]
Definitely odd.  That throw is protected by code that checks to see if the engine mode is Mode::Write or Mode::Append, so there's something funky going on if it thinks that's not the case.  Is the code somewhere we can look at it?

**Reply by hr87** [reply]
Sadly it is LANL internal. One thing I just noticed is that the unit tests uses adios 2.9.2 while the main program used adios 2.8.2. I will try to recompile the main program with 2.9.2 to see if that is an issue.

**Reply by hr87** [reply]
It seems to be a bug (or something) in 2.8.2. Switching to 2.9.2 lets me do a full restart and append to the existing file. 

PS: The documentation does not have anything about the random access read mode. I was able to pull that from the code, but the docs should be updated to include it.

**Reply by eisenhauer** [reply]
Thanks.  The split between Mode::Read and Mode::ReadRandomAccess came only with BP5, and while it got into release notes, it doesn't look like the docs were updated before 2.9 was released.  They were updated before 2.10 though, so current docs at least mention ReadRandomAccess.  I do note that there are other spots where we should have mentioned it and will update in a while.
